flink-connector-pinot/src/test/java/org/apache/flink/streaming/connectors/pinot/PinotTestBase.java

```diff
@@ -85,6 +92,35 @@
                             .withStartupTimeout(Duration.ofSeconds(180))
             );
 
+    @ClassRule
+    public static final MiniCluster MINI_CLUSTER = new MiniCluster(
```

Comment:
Either remove the `@ClassRule` or, my preference, use https://github.com/apache/flink/blob/master/flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MiniClusterWithClientResource.java. Currently the `@ClassRule` has no effect because the `MiniCluster` does not implement the required interfaces therefore you have to call `@BeforeAll` and `@AfterAll` manually.